src/main/java/org/mule/extension/http/api/request/HttpConnectivityValidator.java

```diff
@@ -0,0 +1,202 @@
+  private boolean followRedirects = false;
+
+  /**
+   * Maximum time that the request element will block the execution of the flow waiting for the HTTP response.
+   */
+  @Parameter
+  @Optional(defaultValue = "10000")
+  @Placement(order = 4)
+  private Integer responseTimeout = 10000;
+
+  /**
+   * Response timeout time unit.
+   */
+  @Parameter
+  @Optional(defaultValue = "MILLISECONDS")
+  @Placement(order = 5)
+  private TimeUnit responseTimeoutUnit = MILLISECONDS;
+
+  /**
+   * Request builder to be used to connectivity test.
+   */
+  @ParameterGroup(name = "Request Builder", showInDsl = false)
+  @Placement(order = 6)
+  private HttpRequesterTestRequestBuilder requestBuilder;
+
+  /**
+   * Validation applied to the connectivity test response.
+   */
+  @Parameter
+  @Optional
+  @DisplayName("Response Validator")
+  @Placement(order = 7)
+  @Expression(NOT_SUPPORTED)
+  private ResponseValidator responseValidator;
+
+  private SuccessStatusCodeValidator defaultStatusCodeValidator = new SuccessStatusCodeValidator("0..399");
+
+  @Inject
+  // It's only used to propagate the initialisation to response validator.
+  private MuleContext muleContext;
+
+  public void validate(HttpExtensionClient client, RequestConnectionParams connectionParams)
+      throws ExecutionException, InterruptedException, ResponseValidatorTypedException {
+    HttpRequest request = buildTestRequest(connectionParams);
+    Result<Object, HttpResponseAttributes> result = sendRequest(client, request);
+    validateResult(request, result);
+  }
+
+  private void validateResult(HttpRequest request, Result result) {
+    getResponseValidator().validate(result, request);
+  }
+
+  private Result<Object, HttpResponseAttributes> sendRequest(HttpExtensionClient client, HttpRequest request)
+      throws InterruptedException, ExecutionException {
+    int responseTimeoutInt = (int) responseTimeoutUnit.toMillis(responseTimeout.longValue());
+    HttpResponse response =
+        client.send(request, responseTimeoutInt, followRedirects, resolveAuthentication(client)).get();
+
+    return new HttpResponseToResult()
+        .convert(new VoidHttpRequesterCookieConfig(), muleContext, response, response.getEntity(),
+                 response.getEntity()::getContent,
+                 request.getUri());
+  }
+
+  private HttpRequest buildTestRequest(RequestConnectionParams connectionParams) {
+    String uriString = getUriString(connectionParams);
+    return HttpRequest.builder()
+        .uri(uriString)
+        .method(requestMethod)
+        .headers(toMultiMap(requestBuilder.getRequestHeaders()))
+        .queryParams(toMultiMap(requestBuilder.getRequestQueryParams()))
+        .entity(requestBuilder.buildEntity())
+        .build();
+  }
+
+  private static MultiMap<String, String> toMultiMap(Iterable<? extends KeyValuePair> asList) {
+    MultiMap<String, String> asMultiMap = new MultiMap<>();
+    asList.forEach(pair -> asMultiMap.put(pair.getKey(), pair.getValue()));
```

Comment:
use a Collector instead (`MultiMap#toMultiMap` for instance)

Reply:
That collector is part of the mule-api since 1.5
Removed the method to use an utility one